apps/wallet/src/ui/app/pages/accounts/manage/ManageAccountsPage.tsx

```diff
@@ -1,20 +1,45 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { useMemo } from 'react';
 import { useNavigate } from 'react-router-dom';
 import { AccountGroup } from './AccountGroup';
 
 import Overlay from '../../../components/overlay';
 import { useAccounts } from '../../../hooks/useAccounts';
+import { type AccountSourceType } from '_src/background/account-sources/AccountSource';
+import { type AccountType, type SerializedUIAccount } from '_src/background/accounts/Account';
 
 export function ManageAccountsPage() {
 	const { data: accounts = [] } = useAccounts();
+
 	const navigate = useNavigate();
+	// todo: ideally we would group by account source, but I noticed that zk
+	// accounts don't have an account source entry for some reason...
+	const groupedAccounts = useMemo(() => {
+		return accounts.reduce(
+			(acc, account) => {
+				if (!acc[account.type]) {
+					acc[account.type] = [];
+				}
+				acc[account.type].push(account);
+				return acc;
+			},
+			{} as Record<AccountType, SerializedUIAccount[]>,
+		);
+	}, [accounts]);
 
 	return (
 		<Overlay showModal title="Manage Accounts" closeOverlay={() => navigate('/home')}>
 			<div className="flex flex-col gap-4 flex-1">
-				<AccountGroup accounts={accounts} type="mnemonic-derived" />
+				{Object.entries(groupedAccounts).map(([type, accounts]) => (
+					<AccountGroup
+						key={type}
+						accountSource={''}
```

Comment:
is this empty because ZK accounts don't have an account source? could we add a fixme / todo if this needs re-worked?

Reply:
yea, this needs a `todo:`